staging/src/slime.io/slime/modules/meshregistry/pkg/source/zookeeper/watching.go

```diff
@@ -32,228 +34,445 @@ func (s *Source) ServiceNodeDelete(path string) {
 	}
 }
 
-func (s *Source) ServiceNodeUpdate(services []string, path string) {
-	if !verifyPath(path, false) {
-		return
-	}
-	for _, child := range services {
-		go s.Watch(path+"/"+child, s.ServiceNodeUpdate, nil, s.zkGatewayModel)
-	}
+func (s *Source) EndpointUpdate(provider, consumer []string, path string) {
+	s.handleServiceData(s.cache, provider, consumer, strings.Split(path, "/")[2])
 }
 
 func (s *Source) Watching() {
```

Comment:
Consider a more appropriate name

Reply:
I guess it should be the corresponding name of `Polling()`, I am not going to modify it here.

Reply:
add a comment

Reply:
do it in cleancode task